### PR TITLE
[FIX] Request id no longer exists after concurrency rollback

### PR DESCRIPTION
**Pull request by StefanRijnhart**
The request id stored on the http request can be rolled back when a concurrency error occurs. Upon retry, the id does not exist anymore.

Related logs look like this:
```
Traceback (most recent call last):
  File "/opt/odoo/parts/odoo/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
TransactionRollbackError: could not serialize access due to concurrent update
2017-01-16 05:31:59,837 25491 INFO odoo [-][-][-] openerp.service.model: SERIALIZATION_FAILURE, retry 1/5 in 0.3513 sec...
[...]
IntegrityError: insert or update on table "auditlog_log" violates foreign key constraint "auditlog_log_http_request_id_fkey"
DETAIL:  Key (http_request_id)=(2263) is not present in table "auditlog_http_request".
```

### CODE REVIEW

File: auditlog/models/http_request.py
Comment (sebalix) [inline auditlog/models/http_request.py:5]: One line space between third modules and Odoo imports?

**Review by sebalix — APPROVED** [review]
Minor change, otherwise :+1:

File: auditlog/models/http_request.py
Comment (StefanRijnhart) [reply auditlog/models/http_request.py:5]: Mmm, this is not linted and it's not even in the guidelines. https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports only explicitely mentions the order of the imports, even if it uses newlines to clarify the grouping in the example. Seems like a cornercase to me, no?

Comment (sebalix) [reply auditlog/models/http_request.py:5]: Yes it is, that's why I am not blocking the PR ;) I'm used to do this to clarify the Python dependencies (like the example you linked), and it must be difficult for a quality checker to guess that I think.

**Review by moylop260 — APPROVED** [review]
LGTM
I didn't know AsIs method to avoid sql injection from table section
Thanks!

File: auditlog/models/http_request.py
Comment (moylop260) [reply auditlog/models/http_request.py:5]: In the guidelines recommend us run `isort`
This lint detect this case and auto-fix.
But it is optional.
If you don't want is goos to us

**Comment by sebalix** [comment]
@StefanRijnhart Do you think this error could occurred on 10.0? To know if there is an interest to include it in https://github.com/OCA/server-tools/pull/707

**Comment by StefanRijnhart** [comment]
I would be surprised if it did not occur in 10.0, so please have it included there.

**Comment by sebalix** [comment]
Two approvals, is someone available to merge this PR please ? ;)

**Pull request merged**
